### ISSUE TITLE
Replace db_task_write_lock with a solution that is specific to database backends and removes reliance on cache layers

### ISSUE THREAD

**Issue by rtibbles** [learningequality/kolibri#7891]
### Observed behavior
We have seen various issues with our attempts to use both diskcache and Redis to store a write lock for tasks on the database. This includes the write lock being purged from Redis, and it not being properly turned off from diskcache.

### Expected behavior
We should be able to easily infer whether a task is currently blocked from writing to the database.

For SQLite, this is as simple as attempting to get any sort of write lock on the database within a transaction, and retrying if it fails to do so because of a timeout error.

An implementation of this that would be specific to SQLite could be as simple as this, as any attempt to write to SQLite will result in a write lock:
```
from django.db import transaction
from django.core.exceptions import OperationalError

@contextmanager
def db_task_write_lock():
    while True:
        try:
            with transaction.atomic():
                <some operation that writes to the DB to give a write lock>
                yield
                <revert the above write operation>
                break
            except OperationalError as e:
                if "database is locked" not in str(e):
                    raise e
```

For Postgresql, we would be better off using some sort of advisory lock (potentially copying some or all of the implementation by @bjester here: https://github.com/learningequality/studio/blob/hotfixes/contentcuration/contentcuration/db/advisory_lock.py).

**Comment by iharshit009** [comment]
I would like to work on this issue, Should I proceed @rtibbles ?

**Comment by rtibbles** [comment]
Hi @iharshit009, I think there are a quite a few implementation details to be worked out here, and thinking it through even the sample above is incorrect, so I think it would be better not to work on this at this time.

**Comment by iharshit009** [comment]
Okay, no issues

**Comment by rtibbles** [comment]
Implementing this should also resolve this issue: https://github.com/learningequality/kolibri/issues/7463 as we will no longer use Redis locks in the way that is causing issues here.

**Comment by rtibbles** [comment]
> #7463 as we will no longer use Redis locks in the way that is causing issues here.

Sorry, this is not true - as the locks being used in this case are for caching only, and cannot be replaced by database level locks.

**Comment by rtibbles** [comment]
Fixed in #8004
